packages/shared/dao/pick-realm-algorithm/allPeers.ts

```diff
@@ -0,0 +1,41 @@
+import { Candidate } from "../types"
+import { AlgorithmContext, AlgorithmLink, AlgorithmLinkTypes, AllPeersScoreParameters } from "./types"
+import { usersCount, maxUsers, selectFirstByScore, defaultScoreAddons } from "./utils"
+
+export function allPeersScoreLink({ baseScore, discourageFillTargetPercentage, fillTargetPercentage, latencyDeductionsParameters }: AllPeersScoreParameters): AlgorithmLink {
+  function usersScore(candidate: Candidate) {
+    const count = usersCount(candidate)
+    const max = maxUsers(candidate)
+
+    // We prefer realms that have users. Those will have at least baseScore
+    if (count === 0) return 0
+
+    const linearUsersScore = (users: number) => baseScore + users
+
+    if (max) {
```

Comment:
why can `max` be undefined?

Reply:
Archipelago servers currently don't have maxUsers... That'll probably change, but will also probably change to something so big that this will be not so relevant... We'll see what happens then.